core/rawdb/database_test.go

```diff
@@ -15,3 +15,293 @@
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
 package rawdb
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/crypto"
+	"github.com/ava-labs/libevm/ethdb"
+	"github.com/ava-labs/libevm/log"
+	"golang.org/x/exp/slog"
+)
+
+type stubDatabase struct {
+	ethdb.Database
+	iterator     ethdb.Iterator
+	ancientSizes map[string]uint64
+	ancients     uint64
+	tails        uint64
+}
+
+func (s *stubDatabase) NewIterator(keyPrefix, keyStart []byte) ethdb.Iterator {
+	return s.iterator
+}
+
+// AncientSize is used in [InspectDatabase] to determine the ancient sizes.
+func (s *stubDatabase) AncientSize(kind string) (uint64, error) {
+	size, ok := s.ancientSizes[kind]
+	if !ok {
+		return 0, fmt.Errorf("ancient size for kind %q not found", kind)
+	}
+	return size, nil
+}
+
+// Ancients is used in [InspectDatabase] to determine the ancients counts.
+func (s *stubDatabase) Ancients() (uint64, error) {
+	return s.ancients, nil
+}
+
+// Tail is used in [InspectDatabase] to determine the ancients counts.
+func (s *stubDatabase) Tail() (uint64, error) {
+	return s.tails, nil
+}
+
+func (s *stubDatabase) Get(key []byte) ([]byte, error) {
+	if bytes.Equal(key, trieNodeAccountPrefix) {
+		return nil, nil // NOT path scheme, see [inspectFreezers]
+	}
+	return nil, fmt.Errorf("key not found: %x", key)
+}
+
+func (s *stubDatabase) ReadAncients(fn func(ethdb.AncientReaderOp) error) (err error) {
+	return nil
+}
+
+type stubIterator struct {
+	ethdb.Iterator
+	i        int
+	kvs      []keyValue
+	released bool
+}
+
+type keyValue struct {
+	key   []byte
+	value []byte
+}
+
+func (s *stubIterator) Next() bool {
+	s.i++
+	available := s.i < len(s.kvs)
+	return available
+}
+
+func (s *stubIterator) Release() {
+	s.released = true
+}
+
+func (s *stubIterator) Key() []byte {
+	return s.kvs[s.i].key
+}
+
+func (s *stubIterator) Value() []byte {
+	return s.kvs[s.i].value
+}
+
+type simpleLogger struct {
+	slog.Handler
+}
+
+func (l *simpleLogger) Handle(_ context.Context, r slog.Record) error {
+	attributes := make([]string, 0, r.NumAttrs())
+	r.Attrs(func(attr slog.Attr) bool {
+		attributes = append(attributes, fmt.Sprintf("%s=%v", attr.Key, attr.Value))
+		return true
+	})
+	fmt.Fprintln(os.Stdout, r.Message, attributes)
+	return nil
+}
+
+func (l *simpleLogger) Enabled(_ context.Context, _ slog.Level) bool { return true }
+
+func ExampleInspectDatabase() {
```

Comment:
I need some time to understand what's going on here. It's very complicated for a test.

1. Is every single key really necessary (i.e. why do we care to cover every single switch case)?
2. Is there a way to avoid making the reader do mental arithmetic everywhere? For example, can the `WithDatabaseStatsTransformer()` input argument be used as the `got` value to be tested? In its current form there's little value to this being an example for `godoc` IMO as it's a contrived scenario, so testing another way is acceptable.